### PR TITLE
Remove the documentation Hint "(?)" functionality

### PR DESCRIPTION
**Pull request by fballiano**
This PR should solve https://github.com/OpenMage/magento-lts/issues/1374  https://github.com/OpenMage/magento-lts/issues/1376 and https://github.com/OpenMage/magento-lts/issues/1518 removing links to old magento.com documentation from the legacy backend theme.

I've divided the PR into 4 commits:
- delete the "get help for this page" link as per https://github.com/OpenMage/magento-lts/issues/1376
- delete the "what is this" hint as per https://github.com/OpenMage/magento-lts/issues/1374
- remove all unused PHP code related to the first 2 commits
- remove all unused PHP code related to the generation of "merch.docs.magento.com" links in the "system -> configuration" section of the backend

I've tested the backend after these modifications and it seems to work fine on all sections

### CODE REVIEW

**Comment by kiatng** [comment]
Shouldn't this be merged to 1.9.4.x instead of 20.0?

**Comment by Flyingmana** [comment]
Removing the functions has a small chance of beeing a BC incompatibility.
Also we can backport it later, if it proves to be safe.
Also while the current state is not nice, its not necessarily a bug.

**Review by Flyingmana — APPROVED** [review]

**Comment by fballiano** [comment]
> Removing the functions has a small chance of beeing a BC incompatibility.
> Also we can backport it later, if it proves to be safe.
> Also while the current state is not nice, its not necessarily a bug.

That's absolutely true, that's why I split the commits.
At the same time removing the code from the templates leaving the "php" logic behind seemed kinda wrong.
Also, for sure nobody ever used/extended this feature ehehheh

:-)

**Review by kiatng — APPROVED** [review]

**Comment by fballiano** [comment]
@colinmollenhour what do you think about this?

**Review by colinmollenhour — APPROVED** [review]

File: app/code/core/Mage/Adminhtml/Block/Store/Switcher.php [hunk 1]
```diff
@@ -223,38 +223,4 @@ public function hasDefaultOption($hasDefaultOption = null)
...
-
-    /**
-     * Return url for store switcher hint
-     *
-     * @return string
-     */
-    public function getHintUrl()
-    {
-        if (null === $this->_hintUrl) {
-            $this->_hintUrl = Mage::helper('core/hint')->getHintByCode(self::XPATH_HINT_KEY);
```
Comment (Sekiphp) [inline app/code/core/Mage/Adminhtml/Block/Store/Switcher.php:235]: I think, that we should also remove constant `XPATH_HINT_KEY`, which is currently used nowhere

Comment (fballiano) [reply app/code/core/Mage/Adminhtml/Block/Store/Switcher.php:235]: you're totally right, I have a problem tho, I've deleted that forked repo cause I messed it up and I think I can't modify the PR anymore :-|

Comment (Flyingmana) [reply app/code/core/Mage/Adminhtml/Block/Store/Switcher.php:235]: its a bit "complicated", but you can checkout a PullRequest directly, see step5: https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally

Comment (fballiano) [reply app/code/core/Mage/Adminhtml/Block/Store/Switcher.php:235]: I tried but I think I would be able to do actually update the PR only if I had write access to this repo. If seems (https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally) that I can checkout a PR but then I can only create another one... so if you agree I'd merge this one and I'll create another PR later, just to remove the constant, if that's ok.

Comment (Sekiphp) [reply app/code/core/Mage/Adminhtml/Block/Store/Switcher.php:235]: @fballiano I think, that you should create new PR now (with complete code) and link this pull request inside. This PR should be closed, if is not possible to edit it.

Comment (fballiano) [reply app/code/core/Mage/Adminhtml/Block/Store/Switcher.php:235]: it seems a bit overkill just to remove an unused constant... anyways... I'll do that, let's not get stuck on that.

**Comment by fballiano** [comment]
I'm closing this PR, because it's been superseeded by https://github.com/OpenMage/magento-lts/pull/1536 (as requested by @Sekiphp)

**Pull request closed without merging**